tests/components/scrape/__init__.py

```diff
@@ -113,5 +113,3 @@ async def async_update(self, data: bool | None = True) -> None:
             self.data = "<div class='return'>secret text</div>"
         if self.payload == "test_scrape_sensor_no_data":
             self.data = None
-        if self.count == 3:
```

Comment:
I think this was in there before I got involved, as a way to check how many times if was called.

Reply:
It was a way to deal with refreshes without data before the coordinator was implemented.
So not needed anymore.

Reply:
I think the tests should not be calling the mock three times.
So the limit here shouldn't need to be removed.

Reply:
That should be sorted with the other change you suggested. Can also assert mock_calls to see this is true it's not called three times. 
However this should anyway be taken out/cleaned as it's not used anymore.

Reply:
Possibly... but it helped pinpoint the unnecessary extra calls...
I suggest removing in a follow-up PR instead

Reply:
Happy to remove in this PR if we assert call counts another way.